Automattic/jetpack
# Phpcs: Replace the requirelist with an excludelist

#### Changes proposed in this Pull Request:

This replaces the list of files that are supposed to be clean in phpcs (`bin/phpcs-requirelist.js`) with a list of files that are known to not be clean in phpcs (`bin/phpcs-excludelist.txt`).

The end goal is to have all files being fully checked by phpcs. The progress towards that goal is easier to check when it's "no files are left in bin/phpcs-excludelist.txt" than when it's "all files are covered by bin/phpcs-requirelist.js", and it's easier to find work to do when that work is "clean up a file in excludelist" than "clean up a file that's not in requirelist".

This also makes things more straightforward when adding new files, as the default will now be to fully check the files rather than to not check them but have dangerfile.js try to complain about them.

It also avoids files accidentally falling out of the supposed-to-be-clean list if they're renamed and the requirelist isn't updated.

Plus, the existing requirelist is inaccurate: it claims everything in packages/ is required, but in practice packages/analyzer/ has never passed. This change itself won't do anything directly to prevent that sort of thing in the future, but cleaning it up is necessary if we want to start having CI run phpcs over the supposed-to-be-clean files (which is my next project).

#### Jetpack product discussion


None outside of some private Slack chats.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check that `composer php:requirelist` shows no errors or warnings.

* If you really want, check that running phpcs on the files listed in `bin/phpcs-excludelist.txt` reports at least one warning or error in each one.

#### Proposed changelog entry for your changes:

* Developer only, none needed.
